src/client/mod.rs

```diff
@@ -216,12 +247,29 @@ impl Client {
         })
     }
 
-    pub fn begin(&mut self, n_metrics: u64) -> io::Result<&mut Client> {
+    pub fn begin(&mut self, n_indoms: u64, n_instances: u64, n_metrics: u64) -> io::Result<&mut Client> {
```

Comment:
Also, since last year's program ended, I have thought about multiple times to make indom a part of the implementation and not expose the idea to the user at all, and put all instances of a metric in an autogenerated indom